frontend/src/components/Header/Header.tsx

```diff
@@ -74,28 +75,49 @@ function InstallationPicker(page: string) {
     const { TranslateText } = useLanguageContext()
     const [allPlantsMap, setAllPlantsMap] = useState<Map<string, string>>()
     const { installationCode, switchInstallation } = useInstallationContext()
+    const [showActivePlants, setShowActivePlants] = useState<boolean>(true)
+    const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false)
+
     useEffect(() => {
-        BackendAPICaller.getEchoPlantInfo().then((response: EchoPlantInfo[]) => {
+        const plantPromise = showActivePlants ? BackendAPICaller.getActivePlants() : BackendAPICaller.getEchoPlantInfo()
+        plantPromise.then(async (response: EchoPlantInfo[]) => {
             const mapping = mapInstallationCodeToName(response)
             setAllPlantsMap(mapping)
         })
-    }, [])
+    }, [showActivePlants])
```

Comment:
My understanding is the list of active plants is updated when the checkbox is ticked or unticked. I was considering updating it also when someone clicks in the autocomplete, with the onFocus parameter, what do you think?

Reply:
Sure I don't see why not

Reply:
It does make the code slightly uglier though since we need to add a dummy variable in the onFocus function which can be placed in the list of variables in the useEffect. Unless we want to reuse showActivePlants for that use, but I'm not sure if that's a clean way of doing it